source/postprocess/topography_to_file.cc

```diff
@@ -0,0 +1,290 @@
+
+      for (; cell != endc; ++cell)
+        if (cell->is_locally_owned() && cell->at_boundary())
+          for (unsigned int face_no = 0; face_no < GeometryInfo<dim>::faces_per_cell; ++face_no)
+            if (cell->face(face_no)->at_boundary())
+              {
+                if ( cell->face(face_no)->boundary_id() != relevant_boundary)
+                  continue;
+
+                face_vals.reinit( cell, face_no);
+
+                for (unsigned int corner = 0; corner < face_corners.size(); ++corner)
+                  {
+                    Point<dim> vertex = face_vals.quadrature_point(corner);
+                    double elevation = this->get_geometry_model().height_above_original_surface(vertex);
+                    stored_values.push_back (std::make_pair(vertex, elevation));
+                    if ( elevation > local_max_height)
+                      local_max_height = elevation;
+                    if ( elevation < local_min_height)
+                      local_min_height = elevation;
+                  }
+              }
+
+      //Calculate min/max topography across all processes
+      const double max_topography = Utilities::MPI::max(local_max_height, this->get_mpi_communicator());
+      const double min_topography = Utilities::MPI::min(local_min_height, this->get_mpi_communicator());
+
+      //Write results to statistics file
+      statistics.add_value ("Minimum topography (m)",
+                            min_topography);
+      statistics.add_value ("Maximum topography (m)",
+                            max_topography);
+      //{
+      const char *columns[] = { "Minimum topography (m)",
+                                "Maximum topography (m)"
+                              };
+      for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+        {
+          statistics.set_precision (columns[i], 8);
+          statistics.set_scientific (columns[i], true);
+        }
+      //}
+
+      output_stats.precision(4);
+      output_stats << min_topography << " m, "
+                   << max_topography << " m";
+
+
+
+
+      // Write the solution to file
+
+      // if this is the first time we get here, set the last output time
+      // to the current time - output_interval. this makes sure we
+      // always produce data during the first time step
+      if (std::isnan(last_output_time))
+        {
+          last_output_time = this->get_time() - output_interval;
+        }
+
+      // return if graphical output is not requested at this time
+      if ((this->get_time() < last_output_time + output_interval)
+          && (this->get_timestep_number() != 0))
+        return std::pair<std::string,std::string>();
+
+
+
+
+
+      for (unsigned int i=0; i<stored_values.size(); ++i)
+        {
+          output_file << stored_values[i].first
+                      << ' '
+                      << stored_values[i].second
+                      << std::endl;
```

Comment:
Since you just write them in the order you put them into the vector anyway, would it make sense to immediately write the pair into the ostringstream output_file in line 82? That way you would save the vector of pairs.